pkg/operator/v1helpers/core_getters.go

```diff
@@ -89,7 +101,11 @@ func (g combinedSecretInterface) Get(name string, options metav1.GetOptions) (*c
 }
 
 func (g combinedSecretInterface) List(opts metav1.ListOptions) (*corev1.SecretList, error) {
-	list, err := g.lister.List(labels.Everything())
+	selector, err := listOptionsToSelector(opts)
+	if err != nil {
+		return nil, err
+	}
+	list, err := g.lister.List(selector)
```

Comment:
nit: I know you haven't changed it but I've just noticed that a few lines below we could allocate memory as we know the size of the array instead of appending.

```
ret := &corev1.SecretList{}
for i := range list {
  ret.Items = append(ret.Items, *(list[i].DeepCopy()))
}
```